components/brave_wallet/browser/solana_transaction.cc

```diff
@@ -212,6 +212,43 @@ SolanaTransaction::GetSignedTransactionBytes(
   return transaction_bytes;
 }
 
+// Get serialized and signed transaction given a signature from a hardware
+// wallet. Only supports transactions with one signer.
+absl::optional<std::vector<uint8_t>>
+SolanaTransaction::GetSignedTransactionBytes(
+    const std::vector<uint8_t>& signature_bytes) const {
+  if (signature_bytes.size() > kSolanaSignatureSize)
+    return absl::nullopt;
+
+  // Combine the signature from hardware the message and send
+  std::vector<uint8_t> transaction_bytes;
+
+  std::uint8_t num_signers = message_.GetNumberOfSigners();
+  if (num_signers != 1) {
+    return absl::nullopt;
+  }
+
+  // 1 signer
+  CompactU16Encode(1, &transaction_bytes);
+
+  // Add signature
+  transaction_bytes.insert(transaction_bytes.end(), signature_bytes.begin(),
+                           signature_bytes.end());
+
+  // Add message
+  auto message_bytes = message_.Serialize(nullptr);
+  if (!message_bytes)
+    return absl::nullopt;
+
+  transaction_bytes.insert(transaction_bytes.end(), message_bytes->begin(),
+                           message_bytes->end());
+
+  if (transaction_bytes.size() > kSolanaMaxTxSize)
```

Comment:
I can't think of a case where an empty transaction is valid, since a transaction in this context includes everything (signatures + message). I think we can probably get away without checking transaction_bytes isn't 0 since we check that the signature_bytes included in transaction_bytes is exactly kSolanaSignatureSize (which is > 0).

Reply:
Ok cool, that line of thinking checks out what I'm seeing as well. Thanks for the additional information

Reply:
Sorry, I left this comment on the wrong thread. Meant that to be for the one about doing byte checks. Can we add this check in still.

Reply:
It's not possible to have 0 byte tx here as message_.Serialize won't produce empty message byte arrays.

Reply:
👍🏻 thanks for confirming that @yrliou I wasn't sure if "" was the same as and empty message in C++. I saw that we've got tests for that part already as well, so I'll resolve this one.